x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/rank/textsimilarity/TextSimilarityRankFeaturePhaseRankCoordinatorContext.java

```diff
@@ -235,4 +239,36 @@ private static float normalizeScore(float score) {
         // while negative values (and 0) will be shifted to (0, 1]
         return Math.max(score, 0) + Math.min((float) Math.exp(score), 1);
     }
+
+    private ChunkScorerConfig resolveChunkingSettings(GetInferenceModelAction.Response response) {
+        if (chunkScorerConfig == null) {
+            return null;
+        }
+
+        if (chunkScorerConfig.chunkingSettings() != null) {
+            return chunkScorerConfig;
+        }
+
+        ChunkingSettings endpointSettings = extractChunkingSettingsFromResponse(response);
+
+        if (endpointSettings != null) {
+            return new ChunkScorerConfig(chunkScorerConfig.size(), chunkScorerConfig.inferenceText(), endpointSettings);
+        }
+
+        return new ChunkScorerConfig(
+            chunkScorerConfig.size(),
+            chunkScorerConfig.inferenceText(),
+            ChunkScorerConfig.createChunkingSettings(ChunkScorerConfig.DEFAULT_CHUNK_SIZE)
+        );
+    }
+
+    private ChunkingSettings extractChunkingSettingsFromResponse(GetInferenceModelAction.Response response) {
```

Comment:
This method will not work. If you step through the code, you'll see that the Elastic Reranker endpoint does not have chunking settings attached to it. So at the end of the day, all you're doing is returning the current hard coded chunking settings, and we're not actually doing anything with this change. 

If you look at the PR I linked to from the issue: https://github.com/elastic/elasticsearch/pull/132169 this is how to get the actual token size we want to use for Reranker endpoints. You want to use the `GetRerankerWindowSizeAction` here, _not_ `GetInferenceModelAction`.